SpatialGDK/Source/SpatialGDK/Private/Schema/ChangeInterest.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Schema/ChangeInterest.h"
+
+DEFINE_LOG_CATEGORY(LogChangeInterest);
+
+namespace SpatialGDK
+{
+void ChangeInterestQuery::DebugOutput(const FString& DiffType) const
+{
+	// Minimal output
+	if (Components.Find(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID) != INDEX_NONE)
+	{
+		UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: client auth interest"));
+	}
+	else
+	{
+		UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: client non auth interest"));
+	}
+
+	/*
+	// Verbose output
+	FString ComponentsString;
+	for (const auto& Component : Components)
+	{
+		ComponentsString += FString::Format(TEXT("{0} "), { Component });
+	}
+
+	UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: components %s"), *ComponentsString);
+
+	FString ComponentSetsString;
+	for (const auto& ComponentSet : ComponentSets)
+	{
+		ComponentSetsString += FString::Format(TEXT("{0} "), { ComponentSet });
+	}
+
+	UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: component sets %s"), *ComponentSetsString);
+	*/
+
+	FString EntitiesString;
+	for (const auto& EntityId : Entities)
+	{
+		EntitiesString += FString::Format(TEXT("{0} "), { EntityId });
+	}
+
+	UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: %s %s"), *DiffType, *EntitiesString);
+}
+
+void ChangeInterestRequest::Clear()
+{
+	SystemEntityId = SpatialConstants::INVALID_ENTITY_ID;
+	QueriesToAdd.Empty();
+	QueriesToRemove.Empty();
+	bOverwrite = false;
+}
+
+void ChangeInterestRequest::DebugOutput() const
+{
+	UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: client entity id %lld"), SystemEntityId);
+
+	for (const auto& Query : QueriesToAdd)
+	{
+		Query.DebugOutput(TEXT("ADD"));
+	}
+
+	for (const auto& Query : QueriesToRemove)
+	{
+		Query.DebugOutput(TEXT("REMOVE"));
+	}
+
+	UE_LOG(LogChangeInterest, Log, TEXT("Interest diff: overwrite %d"), bOverwrite);
+}
+
+void ChangeInterestRequest::CreateRequest(Worker_CommandRequest& OutCommandRequest) const
```

Comment:
Why take an out reference if you could return by value?

Reply:
Ahh this is left-over form when the CommandRequest was being passed into the InterestFactory directly, will update.